memory/src/main/java/org/opensearch/ml/memory/action/conversation/CreateInteractionTransportAction.java

```diff
@@ -93,10 +96,9 @@ protected void doExecute(Task task, CreateInteractionRequest request, ActionList
         Integer traceNumber = request.getTraceNumber();
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().newStoredContext(true)) {
             ActionListener<CreateInteractionResponse> internalListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            ActionListener<String> al = ActionListener
-                .wrap(iid -> { internalListener.onResponse(new CreateInteractionResponse(iid)); }, e -> {
-                    internalListener.onFailure(e);
-                });
+            ActionListener<String> al = ActionListener.wrap(iid -> {
```

Comment:
Don't we need to stash the context before creating the interaction? Does `newStoredContext` stashes the context?

Reply:
Not here. If you trace the code, stash is done in the InteractionsIndex through ConversationalMemoryHandler